src/startApp.js

```diff
@@ -52,13 +52,13 @@ const startApp = () => {
       console.time('app2')
       settings.appSettings.version = appMetadata.version
       settings.appSettings.id = appMetadata.identifier
-      app = window[appMetadata.id](
-        settings.appSettings,
-        settings.platformSettings,
-        settings.appData
-      )
-      canvas = app.stage.getCanvas()
-      document.body.appendChild(canvas)
+      return Promise.resolve(
+        window[appMetadata.id](settings.appSettings, settings.platformSettings, settings.appData)
+      ).then(_app => {
+        app = _app
```

Comment:
Just an observation. I would have rather kept app locally scoped to avoid this line here. But there's a `finally` clause in the `getSettings` that relies on `app` existing. I'm not sure though _how_ `getSettings` could ever get a valid `app` object if it's being run BEFORE the app is retrieved from `window[appMetadata.id]`. Could you provide any clarity on that? Are those references to `app` required?

Reply:
yes I see your point, it would be nicer to keep a local scope (and I believe that in an earlier version we actually did that),
but as it is now, the reference to the outer scope `app` is necessary

The `app` variable isn't actually accessed in the finally itself, but in an `onClose` method that is defined there as a platform setting. That `onClose`-callback is the one that ultimately needs a reference to the `app` in order to do some cleanup